components/merkledb/src/views/mod.rs

```diff
@@ -183,37 +183,19 @@ impl<T: RawAccess> View<T> {
     }
 
     fn get_bytes(&self, key: &[u8]) -> Option<Vec<u8>> {
-        if let Some(ref changes) = self.changes.as_ref() {
-            if let Some(change) = changes.data.get(key) {
-                match *change {
-                    Change::Put(ref v) => return Some(v.clone()),
-                    Change::Delete => return None,
-                }
-            }
-
-            if changes.is_cleared() {
-                return None;
-            }
-        }
-
-        self.snapshot().get(&self.address, key)
+        self.changes
```

Comment:
This pattern is encountered multiple times within the code. Can it somehow be encapsulated?

Reply:
It's used 2 times x 2 different variations. I think encapsulation would only increase the amount of code (the necessary function to encapsulate changes would need to include at least 4 params: `snapshot`, `changes`, `name` and `key`), so I didn't do that.

Reply:
Makes sense.